core/CMakeLists.txt

```diff
@@ -138,7 +141,9 @@ if(NOT IOS AND NOT ANDROID)
         timeout_handler_test.cpp
         call_every_handler_test.cpp
         curl_test.cpp
+        # FIXME: Below line is NOT capturing plugin unit tests sources. So adding manually.
         ${plugin_unittest_source_files}
```

Comment:
Isn't it supposed to be extended by the plugins? So in your case, in `plugins/mission/CMakeLists.txt`, you do something like `list(APPEND plugin_unittest_source_files ${CMAKE_CURRENT_SOURCE_DIR}/mission_import_qgc_test.cpp)`?

@julianoes is it how it is meant to be?

Reply:
As per my understanding, `${plugin_unittest_source_files}` should capture unit tests of plugins if we set it in [plugin's CMakeLists.txt](https://github.com/dronecore/DroneCore/pull/235/files#diff-4373271e3de1813cb660c9771ee9ae3cR39). But it was not captured! So I added it manually here.

Reply:
I see that `set` command handles PARENT_SCOPE and [list handles only current scope](https://cmake.org/cmake/help/v3.1/command/list.html). Probably that is why old plugin template had CMake's `set` command usage, but removed now.
Sadly even `set` is not working in my case.

Reply:
Oh wait, but actually this is now in `core`. I don't think that the unit tests in "core" should include unit tests from the plugins.

Question then is whether we want to split the tests per plugin, or if we want to move the `unit_tests_runner` up into the root CMakeLists and use `${UNIT_TEST_SOURCE_FILES}` instead of `${plugin_unittest_source_files}`?

@shakthi-prashanth-m @julianoes what's your opinion on that? For me it's fine to make one unit test executable in the root CMakeLists.txt, I think.

Reply:
Yeah, it makes sense to keep a single unit test executable runner in root CMakeLists.txt. In that case, all plugins should append their unit test source path to `${UNIT_TEST_SOURCE_FILES}`. 
My Question: Why does setting a global variable (using `set` with PARENT_SCOPE) doesn't work in CMake. I've been setting mission plugin unit test and expecting it to reflect in root's another CMake. But not working.

Reply:
If you set it in `plugins/.../CMakeLists.txt` with `PARENT_SCOPE`, it will be available in the parent scope, i.e. `plugins/CMakeLists.txt`. Not in the root CMakeLists, and not in `core/CMakeLists.txt`.

So you're right in that in order to do that, we need to propagate `${UNIT_TEST_SOURCE_FILES}` up with `PARENT_SCOPE` at each level, as described in the documentation [here](https://cmake.org/cmake/help/v3.0/command/list.html):

> The list subcommands APPEND, INSERT, REMOVE_AT, REMOVE_ITEM, REMOVE_DUPLICATES, REVERSE and SORT may create new values for the list within the current CMake variable scope. Similar to the SET command, the LIST command creates new variable values in the current scope, even if the list itself is actually defined in a parent scope. To propagate the results of these operations upwards, use SET with PARENT_SCOPE, SET with CACHE INTERNAL, or some other means of value propagation.

I made a toy example [here](https://ptpb.pw/l3nF).

Reply:
Thank you @JonasVautherin. Its useful.

Reply:
@julianoes are we building both Integration & Unit tests in [root CMakeLists.txt](https://github.com/dronecore/DroneCore/blob/develop/CMakeLists.txt#L134) ?

Reply:
I think propagating the unit tests to the main one makes sense.